drivers/sensor/mhz19b/mhz19b.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2021 G-Technologies Sdn. Bhd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_MHZ19B_MHZ19B
+#define ZEPHYR_DRIVERS_SENSOR_MHZ19B_MHZ19B
+
+#include <kernel.h>
+#include <device.h>
+#include <drivers/uart.h>
+
+#define MHZ19B_BUF_LEN 9
+
+#define MHZ19B_TX_CMD_IDX 2
+#define MHZ19B_RX_CMD_IDX 1
+#define MHZ19B_CHECKSUM_IDX 8
+
+/* Arbitrary max duration to wait for the response */
+#define MHZ19B_WAIT K_SECONDS(1)
+
+enum mhz19b_cmd_idx {
+	/* Command to poll for CO2 */
+	MHZ19B_CMD_IDX_GET_CO2,
+	/* Read range */
+	MHZ19B_CMD_IDX_GET_RANGE,
+	/* Get ABC status */
+	MHZ19B_CMD_IDX_GET_ABC,
+	/* Enable ABC */
+	MHZ19B_CMD_IDX_SET_ABC_ON,
+	/* Disable ABC */
+	MHZ19B_CMD_IDX_SET_ABC_OFF,
+	/* Set detection range to 2000 ppm */
+	MHZ19B_CMD_IDX_SET_RANGE_2000,
+	/* Set detection range to 5000 ppm */
+	MHZ19B_CMD_IDX_SET_RANGE_5000,
+	/* Set detection range to 10000 ppm */
+	MHZ19B_CMD_IDX_SET_RANGE_10000,
+	/* Number of supported commands */
+	MHZ19B_CMD_IDX_MAX,
+};
+
+struct mhz19b_data {
+	/* Max data length is 16 bits */
+	uint32_t data : 16;
+	/* Command buf length is 9, fits inside 4 bits */
+	uint32_t xfer_bytes : 4;
+	/* Boolean */
+	uint32_t data_valid : 1;
+	/* Boolean */
+	uint32_t has_rsp : 1;
```

Comment:
`bool`?

Reply:
Is bit field not okay here?

Reply:
yes, but variable is meant to be used as boolean